lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,336 @@
+    """ Check if the parameters and values in values are valid.  """
+    changed_with_update = False
+
+    for param_value_pair in values:
+        parameter = param_value_pair[0]
+        new_value = param_value_pair[1]
+
+        # check valid modifiable parameters
+        if parameter not in modifiable_params.keys():
+            module.fail_json("%s is not a modifiable parameter. Valid parameters to modify are: %s." % (parameter, modifiable_params.keys()))
+
+        # check allowed datatype for modified parameters
+        str_to_type = {"integer": int, "string": str}
+        if type(new_value) != str_to_type[modifiable_params[parameter][1]]:
+            module.fail_json(msg="%s (type %s) is not an allowed value for the parameter %s. Expected a type %s." %
+                             (new_value, type(new_value), parameter, modifiable_params[parameter][1]))
+
+        # check allowed values for modifiable parameters
+        if str(new_value) not in modifiable_params[parameter][0]:
+            if "-" not in modifiable_params[parameter][0]:
+                module.fail_json(msg="%s is not an allowed value for the parameter %s. Valid parameters are: %s." %
+                                 (value, modified, modifiable_params[parameter][0]))
+
+        # check if a new value is different from current value
+        if str(new_value) != modifiable_params[parameter][2]:
+            changed_with_update = True
+
+    return changed_with_update
+
+def check_changed_parameter_values(values, old_parameters, new_parameters):
+    """ Checking if the new values are different than the old values.  """
+    changed_with_update = False
+
+    # if the user specified parameters to reset, only check those for change
+    if values:
+        for param_value_pair in values:
+            parameter = param_value_pair[0]
+            if old_parameters[parameter] != new_parameters[parameter]:
+                changed_with_update = True
+                break
+    # otherwise check all to find a change
+    else:
+        for parameter in old_parameters.keys():
+            if old_parameters[parameter] != new_parameters[parameter]:
+                changed_with_update = True
+                break
+
+    return changed_with_update
+
+def modify(module, conn, name, values):
+    """ Modify ElastiCache parameter group to reflect the new information if it differs from the current. """
+    # compares current group parameters with the parameters we've specified to to a value to see if this will change the group
+    format_parameters = []
+    for key_value_pair in values:
+        key = key_value_pair[0]
+        value = str(key_value_pair[1])
+        format_parameters.append({'ParameterName': key, 'ParameterValue': value})
+    try:
+        response = conn.modify_cache_parameter_group(CacheParameterGroupName=name, ParameterNameValues=format_parameters)
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
```

Comment:
Be more specific in this message about what was happening when the exception happened, and remember to include the `exception=traceback.format_exc()` argument to fail_json to include the stacktrace.